sim-lib/src/lib.rs

```diff
@@ -276,37 +278,33 @@ impl Simulation {
         Ok(())
     }
 
-    // it validates that the nodes are all on the same network and ensures that we're not running on mainnet.
+    // validates that the nodes are all on the same network and ensures that we're not running on mainnet.
     async fn validate_node_network(&self) -> Result<(), LightningError> {
-        let mut valid_network = Option::None;
-        let unsupported_networks: HashSet<String> = vec!["mainnet", "bitcoin"]
-            .into_iter()
-            .map(|s| s.to_string())
-            .collect();
+        if self.nodes.is_empty() {
+            return Err(LightningError::ValidationError(
+                "we don't control any nodes.".to_string(),
+            ));
+        }
+        let mut running_network = Option::None;
 
         for node in self.nodes.values() {
-            let network = node.lock().await.get_info().network.clone();
-            if unsupported_networks.contains(&network) {
+            let network = node.lock().await.get_network().await?;
+            if network == Network::Bitcoin {
```

Comment:
Ran into this issue in the original PR - LND returns "mainnet" and LDK uses "bitcoin" so we need to check both values here.

Reply:
Oh damn, I thought `from_core_arg` considered "mainnet" and "testnet" as valid variants, but now that I remember some contributors nitpicked that core uses "main" and "test", so we cannot use that.

I'll replace it with `from_str` which allows "testnet", and will replace "mainnet" for "bitcoin" in place.